### ISSUE TITLE
TypeError: unsupported operand type(s) for +: 'int' and 'NoneType'

### ISSUE THREAD

**Issue by ProfessorVenkman** [Electron-Cash/Electron-Cash#939]
Crash Report
============

This crash report was reported through the automatic crash reporting system 🤖

Traceback
--------------

```Python traceback
  File "Electron_Cash-3.3.2-py3.6.egg/electroncash_gui/qt/transaction_dialog.py", line 139, in update_tx_if_in_wallet
  File "Electron_Cash-3.3.2-py3.6.egg/electroncash_gui/qt/transaction_dialog.py", line 210, in update
  File "Electron_Cash-3.3.2-py3.6.egg/electroncash/transaction.py", line 693, in get_fee
  File "Electron_Cash-3.3.2-py3.6.egg/electroncash/transaction.py", line 687, in input_value

TypeError: unsupported operand type(s) for +: 'int' and 'NoneType'
```

Reporter
------------

This issue was reported by 1 user(s):

| Electron Cash Version  | Python Version | Operating System  | Wallet Type  | Locale |
|---|---|---|---|---|
| 3.3.2  | 3.6.4 (default, Nov 15 2018, 17:35:53)  [GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.11.45.5)] | Darwin-17.7.0-x86_64-i386-64bit | imported_privkey | en_US | 

Additional Information
------------------------

The reporting user(s) did not provide additional information.

**Comment by cculianu** [comment]
I produced this exceptions.

Steps to reproduce:

1. Loaded a signed tx from a file.
2. Below is the tx.

```
{
    "hex": "0200000001985e9d6ae304010fbbd705bdceddaccdd54a83acfd44b56234bee07af2568335000000006b483045022100e11eb8be9fccb8dd8db8aa386836aed2a40c579ab3f01d8c973d227d1554856502206c8ff9682a649861803d2d78b097d945f5ba53d4c07e3646ff53a79494bd101141210230ae837a1e14b0cbc7fc554f23f16e991b371c8e463791bcc1bdce8ffaa3e554feffffff02d460b600000000001976a914329a8b9221f7383bf65b0b80ccd73d170d34359588ace8030000000000001976a9146fd365df5870da070e1bfd55ec73f232dadcbd2788aced7e0800",
    "complete": true,
    "final": true,
    "input_values": [
        null
    ]
}
```

**Comment by cculianu** [comment]
Note that I investigated this a little bit and it's due to the test `tx in wallet.transactions` returning true, but then when the wallet tries to compute the tx inputs/outputs it ends up getting a `None` member in one of the lists.

I think to trigger this bug one needs to switch chains... where tx's drop out and reappear on the various chains.  It's definitely related to the fork.

So some of the wallet logic has a possible race condition or bookkeeping error.  I'll have to look at the code path for switching chains and see if tx's disappear "properly" or if they can stick around in an undefined or half-defined state.

This looks like a race condition to me...

**Comment by cculianu** [comment]
Fixed by #942

**Comment by markblundeberg** [comment]
was in contact with second reporter of this bug ("This issue was reported by 2 user(s):"), can confirm it was identical problem.

**Comment by cculianu** [comment]
3.4.1 is perhaps misleadingly named.  

It uses code from 3.3.1 and before.  It's not being kept in synch as much as it should with regular Electron Cash.

This issue is most definitely fixed in 3.3.2 git.
